lib/matplotlib/contour.py

```diff
@@ -1023,8 +1029,9 @@ def _process_contour_level_args(self, args, z_dtype):
                 levels_arg = 7  # Default, hard-wired.
         else:
             levels_arg = self.levels
+
         if isinstance(levels_arg, Integral):
-            self.levels = self._autolev(levels_arg)
+            self.levels = self._autolev(levels_arg, using_default=self.levels is None)
```

Comment:
AFAICS this is the only usage of `_autolev()`. I find the logic of `using_default` a bit hard to follow (in particular also because whether it takes effect depends on `self.logscale`). I suggest to pull the logic in to have `if self.levels is None` instead of `if using_default` inside `_autolev()`.

Reply:
Thanks for bringing this up - I just stared at the code and got confused myself about the logic. I've refactored slightly, by allowing `N = None` through to `_autolev`, and pushing the hard coding of `N = 7` to `_autolev`. I think this makes more sense now, but it would be good to hear what you think about the latest iteration.